app/Repositories/User/UserRepository.php

```diff
@@ -174,6 +178,104 @@ public function updateDefectOfUser($userId, $defectAttachmentId,$requestDefect)
             throw new \Exception('reports.not_updated');
         }
     }
+    /**
+    * in case of no results to user
+    * 
+    * @return $result[]
+    */
+    public function emptyResult()
+    {
+        //get EGP from config
+        $currency[]="0 ".$this->statistics['Currency'];
+        //return zero's of result
+        return $result=[$currency,'0 (0 Days)','0 of 0'];
+    }
+
+    /**
+    * Get all bonuses of user within that month
+    * @param $dateStart
+    * @param $dateEnd
+    * @param $bonusesTotal
+    * return $result[0]
+    */
+    public function bonusesOfUser($dateStart,$dateEnd,$bonusesTotal)
+    {
+        //return zero's of result
+        $result=$this->emptyResult();
+
+        $bonusesTotal=Auth::User()->bonuses()->where('created_at','>=',$dateStart)
```

Comment:
Do not restrict repositories to session users, pass user ID or set model before!